nucleus/metrics/cuboid_metrics.py

```diff
@@ -147,13 +161,15 @@ def eval(
         iou_3d_metric, iou_2d_metric = detection_iou(
             predictions,
             annotations,
-            threshold_in_overlap_ratio=self.iou_threshold,
+            self.iou_threshold,
+            self.enforce_label_match,
         )
 
-        weight = max(len(annotations), len(predictions))
         if self.iou_2d:
+            weight = len(iou_2d_metric)
             avg_iou = iou_2d_metric.sum() / max(weight, sys.float_info.epsilon)
         else:
+            weight = len(iou_3d_metric)
             avg_iou = iou_3d_metric.sum() / max(weight, sys.float_info.epsilon)
```

Comment:
if the `weight` is zero by some chance (is that even possible?), won't that make `avg_iou` absurdly large since we would dividing by `sys.float_info.epsilon`?

Perhaps if `weight == 0` then `avg_iou` should be 0 too?

Not sure what is correct here.

Reply:
good catch - I agree with you the boundary condition is incorrect here, the iou should be zero with weight zero in that case

I will also add unit test coverage for this case